src/p2f.c

```diff
@@ -250,19 +271,27 @@ void flow_record_list_free (joy_ctx_data *ctx) {
  */
 static int flow_key_is_eq (const flow_key_t *a,
                            const flow_key_t *b) {
+    int diff = 0;
+
     if (a->prot != b->prot) {
         return 1;
     }
+
     if (a->sp != b->sp) {
         return 1;
     }
+
     if (a->dp != b->dp) {
         return 1;
     }
-    if (a->sa.s_addr != b->sa.s_addr) {
+
+    memcmp_s(&a->sa.v6_sa, sizeof(struct in6_addr), &b->sa.v6_sa, sizeof(struct in6_addr), &diff);
```

Comment:
I think we need to check for ==EOK before checking diff when using memcmp_s

Reply:
will do

Reply:
err = memcmp_s(&a->sa.v6_sa, sizeof(struct in6_addr), &b->sa.v6_sa, sizeof(struct in6_addr), &diff);
    if ((err != EOK) || (diff != 0)) {
        return 1;
    }